### ISSUE TITLE
Making modules/summon.py Gmail compatible

### ISSUE THREAD

**Issue by miggyb** [cjones/madcow#4]
Here's a short one:

summon.py wasn't working with a GMail account and giving this error:

```
[2012/10/24 10:37:15 - ERROR] problem with module: args=(u'miggyb', u'zomg')
[2012/10/24 10:37:15 - ERROR] Traceback (most recent call last):
[2012/10/24 10:37:15 - ERROR]   File "/usr/local/lib/python2.7/site-packages/madcow/util/__init__.py", line 114, in get_response
[2012/10/24 10:37:15 - ERROR]     return self.response(nick, args, kwargs)
[2012/10/24 10:37:15 - ERROR]   File "/usr/local/lib/python2.7/site-packages/madcow/modules/summon.py", line 32, in response
[2012/10/24 10:37:15 - ERROR]     smtp.login(settings.SMTP_USER, settings.SMTP_PASS)
[2012/10/24 10:37:15 - ERROR]   File "/usr/local/lib/python2.7/smtplib.py", line 555, in login
[2012/10/24 10:37:15 - ERROR]     raise SMTPException("SMTP AUTH extension not supported by server.")
[2012/10/24 10:37:15 - ERROR] SMTPException: SMTP AUTH extension not supported by server.
```

Turns out GMail is kind of quirky and expects TLS but doesn't advertise it, according to this post: http://stackoverflow.com/questions/6123072/smtp-auth-extension-trouble-with-python

Fix that worked for me is the 6th comment in this post:
http://www.nixtutor.com/linux/send-mail-through-gmail-with-python/

Or, back in your code:

```
    smtp = SMTP(settings.SMTP_SERVER)

    smtp.ehlo()
    smtp.starttls()
    smtp.ehlo()
    if settings.SMTP_USER and settings.SMTP_PASS:
        smtp.login(settings.SMTP_USER, settings.SMTP_PASS)
```

TL;DR: smtp.ehlo()
